src/pystatis/table.py

```diff
@@ -75,80 +99,89 @@ def prettify_table(data: pd.DataFrame, db_name: str) -> pd.DataFrame:
         """
         match db_name:
             case "genesis":
-                pretty_data = Table.parse_genesis_table(data)
+                pretty_data = Table.parse_genesis_table(data, language)
             case "zensus":
-                pretty_data = Table.parse_zensus_table(data)
+                pretty_data = Table.parse_zensus_table(data, language)
             case "regio":
-                pretty_data = Table.parse_regio_table(data)
+                pretty_data = Table.parse_regio_table(data, language)
             case _:
                 pretty_data = data
 
         return pretty_data
 
     @staticmethod
-    def parse_genesis_table(data: pd.DataFrame) -> pd.DataFrame:
+    def parse_genesis_table(data: pd.DataFrame, language: str) -> pd.DataFrame:
         """Parse GENESIS table ffcsv format into a more readable format"""
+
+        column_name_dict = COLUMN_NAME_DICT["genesis"][language]
+
         # Extracts time column with name from first element of Zeit_Label column
-        time = pd.DataFrame({data["Zeit_Label"].iloc[0]: data["Zeit"]})
+        time = pd.DataFrame({data[column_name_dict["time_label"]].iloc[-1]: data[column_name_dict["time"]]})
 
         # Extracts new column names from first values of the Merkmal_Label columns
         # and assigns these to the relevant attribute columns (Auspraegung_Label)
-        attributes = data.filter(like="Auspraegung_Label")
-        attributes.columns = data.filter(like="Merkmal_Label").iloc[0].tolist()
+        attributes = data.filter(like=column_name_dict["variable_level"])
+        attributes.columns = data.filter(like=column_name_dict["variable_label"]).iloc[-1].tolist()
 
         # Selects all columns containing the values
         values = data.filter(like="__")
 
         # Given a name like BEV036__Bevoelkerung_in_Hauptwohnsitzhaushalten__1000
         # extracts the readable label and omit both the code and the unit
-        values.columns = [name.split("__")[1] for name in values.columns]
+        values.columns = [re.split(r"_{2,}", name)[1] for name in values.columns]
```

Comment:
why was this changed? This is a change of the logic not related to the language, or is it related to the language problem? We should not change core logic without a good reason. Did you encounter a case where there are more than two underscores?

Reply:
Yes, some tables have multiple consecutive underscores (e.g. 81000-0001). With the old version, this led the column names to be empty strings.